### ISSUE TITLE
😿 Unable to create an image off a custom base image

### ISSUE THREAD

**Issue by PureKrome** [dotnet/sdk-container-builds#403]
Hi lovely team.

So I'm trying to create a simple docker image using `dotnet publish`. There's one catch (isn't there _always_ just one catch?) that this image will contain both my NET7 minimal api -and also- have NewRelic "stuff" magically installed into the image _also_!

So to do this I thought:
- Create a base image, which is just `FROM mcr.microsoft.com/dotnet/aspnet:7.0` (notice this is the final/release version of the image) and then added the New Relic stuff, into this image.
- Now, do `dotnet publish -c release --arch x64 --os linux /t:PublishContainer -p:ContainerImageName=jussy-web-test -p:ContainerBaseImage=jussy-base-test:latest` I get this funky error message:

>C:\Users\justi\.nuget\packages\microsoft.net.build.containers\0.4.0\build\Microsoft.NET.Build.Containers.targets(195,5): error MSB4018: The "CreateNewImage" task failed
unexpectedly. [C:\Projects\<snip>.csproj]
C:\Users\justi\.nuget\packages\microsoft.net.build.containers\0.4.0\build\Microsoft.NET.Build.Containers.targets(195,5): error MSB4018: System.NotImplementedException: C
ONTAINER2003: The manifest for jussy-base-test:latest from registry https://docker.io/ was an unknown type: text/html. Please raise an issue at https://github.com/dotnet/sdk-
container-builds/issues with this message. [C:\Projects\<snip>.csproj]

This was the dockerfile fileI used to create my base image, which I [copied from the NewRelic documentation](https://docs.newrelic.com/install/dotnet/?deployment=linux&docker=yesDocker):

```
# The runtime tag version should match the SDK tag version
FROM mcr.microsoft.com/dotnet/aspnet:7.0 AS final

# Install the agent
RUN apt-get update && apt-get install -y wget ca-certificates gnupg \
&& echo 'deb http://apt.newrelic.com/debian/ newrelic non-free' | tee /etc/apt/sources.list.d/newrelic.list \
&& wget https://download.newrelic.com/548C16BF.gpg \
&& apt-key add 548C16BF.gpg \
&& apt-get update \
&& apt-get install -y newrelic-dotnet-agent

# Enable the agent
ENV CORECLR_ENABLE_PROFILING=1 \
CORECLR_PROFILER={36032161-FFC0-4B61-B559-F6C5D41BAE5A} \
CORECLR_NEWRELIC_HOME=/usr/local/newrelic-dotnet-agent \
CORECLR_PROFILER_PATH=/usr/local/newrelic-dotnet-agent/libNewRelicProfiler.so \
NEW_RELIC_LICENSE_KEY=INSERT_NEW_RELIC_LICENSE_KEY \
NEW_RELIC_APP_NAME=INSERT_YOUR_APP_NAME
```

then did: `docker build --tag jussy-base-test .` and it built this docker image and then placed it locally into my localhost docker desktop.

Is this a legit bug -or- I've just done things wrong?

--- 

Summary simple repo:

- create the NewRelic base image with no NET7 code. (docker build ..)
- dotnet new minimal api
- dotnet publish with the base image being the one from step-1.

**Comment by baronfel** [comment]
Hi @PureKrome - can you try with the ContainerBaseImage set to `registry-1.docker.io/jussy-base-test:latest`? I _think_ that should unblock you.  The default docker url we use in this case should be updated to interact with the raw Docker Registry domains instead of `docker.io` - that domain is not a valid registry API. We have to 'quirk' around this when pushing 
(See our docs on this [here](https://github.com/dotnet/sdk-container-builds/blob/main/docs/RegistryAuthentication.md#docker-hub)) but we should try to detect and compensate for this so that you don't have to specify the correct domain.

**Comment by PureKrome** [comment]
👋🏻 G'Day @baronfel ! Thanks for the prompt reply. Ok, I just tried that and got this (bigger) error:

> C:\Users\justi\.nuget\packages\microsoft.net.build.containers\0.4.0\build\Microsoft.NET.Build.Containers.targets(195,5): error MSB4018: The "CreateNewImage" task failed
unexpectedly. [C:\Projects\<snip>.csproj]
C:\Users\justi\.nuget\packages\microsoft.net.build.containers\0.4.0\build\Microsoft.NET.Build.Containers.targets(195,5): error MSB4018: Microsoft.NET.Build.Containers.Cr
edentials.CredentialRetrievalException: CONTAINER1008: Failed retrieving credentials for "registry-1.docker.io": Failed to execute 'C:\Program Files\Docker\Docker\resour
ces\bin\docker-credential-desktop.EXE get': [C:\Projects\<snip>.csproj] <the rest all snipped>

**Comment by baronfel** [comment]
@PureKrome can you also run `docker login registry-1.docker.io` before this as well?

**Comment by PureKrome** [comment]
@baronfel hold on .. is this for a container that is up in a cloud registry? My container is localhost registry.

When i type that command i get this asked for a `username:` ???

**Comment by baronfel** [comment]
Oh - I think you're hitting https://github.com/dotnet/sdk-container-builds/issues/195 then. We don't currently support using a local image as the base image, I just got confused by your error report because of the initial error message in the report.

**Comment by PureKrome** [comment]
Yeah - looks like it :( dang nabit!

Besides _that issue_ .. did my scenario/use-case make sense? How we want to have a simple .NET 7 minimal api .. which we will host in a docker container .. but _that_ container also needs to send all metrics to New Relic .. which means we need to have some software installed into the container running the web api/app. And to get around this, I'm using a base image with the N.R. software installed .

**Comment by baronfel** [comment]
Definitely makes sense - and it's a use case we _want_ to support and just have a gap for. We haven't coded how to _load_ an image from a local daemon, only save an image to the daemon.

**Comment by baronfel** [comment]
I'm going to close this as a duplicate of #195 then - sorry for the inconvenience!
